src/prommis/nanofiltration/costing/diafiltration.py

```diff
@@ -0,0 +1,713 @@
+        streams={
+            "retentate": {
+                "property_package": m.fs.properties,
+                "side_streams": [10],
+                "has_energy_balance": False,
+                "has_pressure_balance": False,
+            },
+            "permeate": {
+                "property_package": m.fs.properties,
+                "has_feed": False,
+                "has_energy_balance": False,
+                "has_pressure_balance": False,
+            },
+        },
+    )
+
+    assert isinstance(
+        m.fs.stage3, MSContactor
+    )  # check that stage3 exists and is an MSContactor
+
+    # Retentate side checks
+    assert hasattr(
+        m.fs.stage3, "retentate_inlet_state"
+    )  # check that there is a retentate feed
+    assert hasattr(
+        m.fs.stage3, "retentate_side_stream_state"
+    )  # check that a side stream exists
+    for k in m.fs.stage3.retentate_side_stream_state:
+        assert k == (0, 10)  # check that the side stream only exists at element 10
+    assert not hasattr(
+        m.fs.stage3, "retentate_energy_balance"
+    )  # check that there are no energy balances
+    assert not hasattr(
+        m.fs.stage3, "retentate_pressure_balance"
+    )  # check that there are no pressure balances
+
+    # Permeate side checks
+    assert not hasattr(
+        m.fs.stage3, "permeate_inlet_state"
+    )  # check that there is no permeate feed
+    assert not hasattr(
+        m.fs.stage3, "permeate_side_stream_state"
+    )  # check that there are no side streams on permeate side
+    assert not hasattr(
+        m.fs.stage3, "permeate_energy_balance"
+    )  # check that there are no energy balances
+    assert not hasattr(
+        m.fs.stage3, "permeate_pressure_balance"
+    )  # check that there are no pressure balances
+
+    m.fs.stage3.length = pyo.Var(units=pyo.units.m)
+    m.fs.stage3.length.fix(10)
+
+    m.fs.stage3.solvent_flux = pyo.Constraint(
+        m.fs.stage3.elements,
+        rule=solvent_rule,
+    )
+    m.fs.stage3.solute_sieving = pyo.Constraint(
+        m.fs.stage3.elements,
+        m.fs.solutes,
+        rule=stage3_solute_rule,
+    )
+
+    # Add second feed mixer
+    m.fs.mix2 = Mixer(
+        num_inlets=2,
+        property_package=m.fs.properties,
+        material_balance_type=MaterialBalanceType.componentTotal,
+        energy_mixing_type=MixingType.none,
+        momentum_mixing_type=MomentumMixingType.none,
+    )
+
+    # Add 3 Arcs to connect third stage to remainder of flowsheet
+    m.fs.stream4 = Arc(
+        source=m.fs.stage2.permeate_outlet,
+        destination=m.fs.mix2.inlet_2,
+    )
+    m.fs.stream5 = Arc(
+        source=m.fs.mix2.outlet,
+        destination=m.fs.stage3.retentate_inlet,
+    )
+    m.fs.stream6 = Arc(
+        source=m.fs.stage3.retentate_outlet,
+        destination=m.fs.mix1.inlet_1,
+    )
+
+    pyo.TransformationFactory("network.expand_arcs").apply_to(m)
+
+    # Unfix Feed to Mixer 1 inlet 1
```

Comment:
It is often also helpful to have a separate setting the variables from the build method too - for one this makes it easier for another user to take your build method but apply their own design conditions (also part of why it is helpful to separate the initialization from the build).